analyses/compare-gistic/util/GISTIC-comparison-functions.R

```diff
@@ -0,0 +1,345 @@
+  # read in and tidy this data into a data.frame that contains
+  # the amplified/deleted genes, the detection peak, and
+  # the IDs of the samples in the corresponding peaks along with
+  # with their CN status call.
+  #
+  # Args:
+  #   all_lesions_file: file path to GISTIC's `all_lesion.conf_90.txt` file
+  #   amp_genes_file: file path to GISTIC's `amp_genes.conf_90.txt` file
+  #   del_genes_file: file path to GISTIC's `del_genes.conf_90.txt` file
+  #   output_filename: string to save the output file as
+  #
+  # Return:
+  #   final_df: data.frame with the relevant data from the `all_lesions`,
+  #             `amp_genes`, and `del_genes` files
+  
+  # Read in `all_lesions_file`
+  gistic_all_lesions_df <- data.table::fread(all_lesions_file,
+                                             data.table = FALSE)
+  
+  # Run `get_genes_object` function on `amp_genes` and `del_genes` files to get
+  # a data.frame with just the genes and their corresponding detection peak
+  amp_genes_df <- get_genes_object(amp_genes_file,
+                                   include_peak_info = TRUE)
+  
+  del_genes_df <- get_genes_object(del_genes_file,
+                                   include_peak_info = TRUE)
+  
+  # Bind the rows from the above data.frames into one data.frame
+  final_df <- rbind(amp_genes_df, del_genes_df)
+  
+  
+  # Wrangle the GISTIC all lesions data to be in a comparable format with our CN calls
+  gistic_all_lesions_df <- gistic_all_lesions_df %>%
+    dplyr::select(
+      -c(
+        "Descriptor",
+        "Peak Limits",
+        "Region Limits",
+        "q values",
+        "Residual q values after removing segments shared with higher peaks",
+        "Broad or Focal",
+        "Amplitude Threshold"
+      )
+    ) %>%
+    as.data.frame() %>%
+    tidyr::gather("Kids_First_Biospecimen_ID",
+                  "status",
+                  -`Unique Name`,
+                  -`Wide Peak Limits`) %>%
+    arrange(desc(`Wide Peak Limits`)) %>%
+    dplyr::mutate(status = dplyr::case_when(status < 0 ~ "loss",
+                                            status > 0 ~ "gain",
+                                            status == 0 ~ "neutral")) %>%
+    dplyr::distinct()
+  
+  # Keep just the range information in the `Wide Peak Limits` column
+  # (In order to merge with the amp/del genes data.frame)
+  gistic_all_lesions_df$`Wide Peak Limits` <-
+    gsub("[(].*", "", gistic_all_lesions_df$`Wide Peak Limits`)
+  
+  # Keep just the detection peak name
+  gistic_all_lesions_df$`Unique Name` <-
+    gsub(" -.*", "", gistic_all_lesions_df$`Unique Name`)
+  
+  # Merge the data from the `all_lesions.conf_90.txt` file with the amp/del gene
+  # data.frame prepped above
+  final_df <- final_df %>%
+    dplyr::left_join(gistic_all_lesions_df, by = c("peak_region" = "Wide Peak Limits")) %>%
+    dplyr::select(gene_symbol, Kids_First_Biospecimen_ID, status, detection_peak = `Unique Name`) %>%
+    dplyr::distinct()
```

Comment:
Can you also comment on this `distinct()` without looking at the data I don't know what's happening.